src/Main.java

```diff
@@ -252,5 +250,16 @@ static void initialArrays() {
         }
     }
 
+    static void truncateBooksArrayOnDeletion() {
+       
+        if (bookQuantity < books.length / 2) {
```

Comment:
Neden kitap dizisi yarıya indi

Reply:
Yönergede belirli bir eşiğin altına düşerse kısaltma işlemi yapın diyordu. Ben de yarıya indiğinde diziyi küçülttüm.

Reply:
1 kitap sildiğimizde yeni bir dizi oluşturalım o silinen kitap harici diğer kitapları tekrardan ekleyelim